internal/nginx-meshctl/commands/services.go

```diff
@@ -96,3 +110,49 @@ func GetServices() *cobra.Command {
 
 	return cmd
 }
+
+// getEndpoints returns a slice of upstream addresses for a service.
+func getEndpoints(ctx context.Context, k8sClient client.Client, svc v1.Service) ([]string, error) {
+	endpointSlices := &discoveryv1beta1.EndpointSliceList{}
+	var upstreamAdresses []string
+	opt := client.MatchingLabels{"kubernetes.io/service-name": svc.Name}
+	if err := k8sClient.List(ctx, endpointSlices, opt); err != nil {
+		fmt.Printf("error getting list of endpoint slices for service: %v\n", err)
+		return nil, err
+	}
+	for _, epSlice := range endpointSlices.Items {
+		epSlice := epSlice
+		if epSlice.Namespace == svc.Namespace {
+			for _, endpoint := range epSlice.Endpoints {
+				upstreamAdresses = append(upstreamAdresses, endpoint.Addresses...)
+			}
+			return upstreamAdresses, nil
+		}
+	}
+	// in the case that a service has no upstreams yet but is in a namespace where injection would be enabled
+	// return nothing rather than an error since that service is still 'part' of the mesh.
+	return nil, nil
+}
+
+// isNamespaceInjectionEnabled returns whether a given namespace has injection enabled.
+func isNamespaceInjectionEnabled(ctx context.Context, k8sClient client.Client, ns string) (bool, error) {
+	meshClient, err := mesh.NewMeshClient(initK8sClient.Config(), meshTimeout)
+	if err != nil {
+		return false, fmt.Errorf("failed to get mesh client: %w", err)
+	}
+	meshConfig, err := GetMeshConfig(meshClient)
+	if err != nil {
+		return false, fmt.Errorf("unable to get mesh config: %w", err)
+	}
+	nsObj := &v1.Namespace{}
+	if err := k8sClient.Get(ctx, client.ObjectKey{
+		Namespace: "",
+		Name:      ns,
+	}, nsObj); err != nil {
+		fmt.Printf("error getting namespace: %v\n", err)
+		return false, err
+	}
+	return slices.Contains(*meshConfig.EnabledNamespaces, ns) ||
+		nsObj.GetLabels()[injectionLabel] == "enabled" ||
```

Comment:
"enabled" is also a constant that we have defined.

Reply:
Didn't see this defined as a constant anywhere in the `mesh` package, so I added a constant definition for it and used that

Reply:
Oh you're right, that's actually in our control plane repo.